docs/JMS-streamingsource.md

```diff
@@ -0,0 +1,102 @@
+# JMS Streaming Source
+
+
+Description
+-----------
+Consumes JMS messages of different types as Message, Text, Bytes and Map from a specified Queue or Topic.
+
+Use Case
+--------
+Use this JMS Source plugin when you want to consume messages from a JMS Queue/Topic and write them to a Table. 
+
+
+Properties
+----------
+**Connection Factory**: Name of the connection factory. If not specified, the value *ConnectionFactory* is considered by
+ default.
+
+**JMS Username**: Username to connect to JMS. This property is mandatory.
+
+**JMS Password**: Password to connect to JMS. This property is mandatory.
+
+**Provider URL**: Provider URL of the JMS Provider. This property is mandatory. For example for the *ActiveMQ* provider
+ you can set: `tcp://hostname:61616`.
+
+**Type**: Queue or Topic. Queue is considered by default.
+
+**Destination**: Queue/Topic name.
```

Comment:
1. **Destination** is not an appropriate name here.  I'll change this. Since I've got only one config class defined for both source and sink, I'll rename this to **Name** and indicate in the description what this means.

2. This property is added by me. I've written a couple of open issues in the engineering doc attached in this PR's description and this is one of them. I've emailed this issue to Bhooshan and Bhooshan CC-ed Chai Pydimukkala on that email. These issue together with some other [engineering doc] (https://docs.google.com/document/d/1ZpXn_tWk71eVJ_A2x5zUZGP5thA7S7uV) issues haven't been addressed yet. 

Since you want to read from a topic/queue, you should provide its name to the plugin in order to do so. That's why I suggested adding this new param.